src/Tribe/Views/V2/Views/Day_View.php

```diff
@@ -146,4 +146,50 @@ protected function build_url_for_date( $url_date, $canonical, array $passthru_va
 
 		return $url;
 	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function setup_template_vars() {
+
+		$template_vars = parent::setup_template_vars();
+		$sorted_events = $this->sort_events( $template_vars['events'] );
+
+		$template_vars['events'] = $sorted_events;
+
+		return $template_vars;
+	}
+
+	/**
+	 * Add timeslot and sort events for the day view.
+	 *
+	 * Iterate over the day events to add timeslots and sort them.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $events  An array of events.
+	 *
+	 * @return array The sorted and modified array.
+	 */
+	protected function sort_events( $events ) {
+
+		$all_day = [];
+		$ongoing = [];
+		$hourly  = [];
+
+		foreach ( $events as $i => $event ) {
+			if ( ! empty( $event->all_day ) ) {
+				$event->timeslot = 'all_day';
+				$all_day[ $i ]   = $event;
+			} elseif ( ! empty( $event->multiday ) ) {
+				$event->timeslot = 'multiday';
+				$ongoing[ $i ]   = $event;
+			} else {
+				$hourly[ $i ] = $event;
```

Comment:
I see that here we don't have any `timeslot` we should have the variable at least set to something to avoid notices on the frontend in case we don't check the variable with `empty` or `isset`